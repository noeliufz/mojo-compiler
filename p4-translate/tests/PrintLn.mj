proc puts(a: Address): int;

proc println(t: Text) {
  puts(Loophole(t, ref Address)^);
}

{
  println("Hello Worldâœ…");
}
