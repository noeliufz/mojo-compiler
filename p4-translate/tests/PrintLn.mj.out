	.text
PrintLn.println:
EXP(
 CALL(
  NAME puts,
  CONST 0,
  MEM(
   ESEQ(
    MOVE(
     TEMP t.0,
     TEMP PrintLn.println.t),
    BEQ(
     TEMP t.0,
     CONST 0,
     PrintLn.println.badPtr, L.1),
    LABEL L.1,
    TEMP t.0), 
   CONST 0, 8, false))),
LABEL L.0


	.data
	.balign 8
L.2:
	.byte 72
	.byte 101
	.byte 108
	.byte 108
	.byte 111
	.byte 32
	.byte 87
	.byte 111
	.byte 114
	.byte 108
	.byte 100
	.byte -30
	.byte -100
	.byte -123
	.byte 0
	.data
	.balign 8
L.3:
	.quad L.2
	.quad 15
	.text
PrintLn:
EXP(
 CALL(
  NAME PrintLn.println,
  CONST 0,
  NAME L.3))


